### ISSUE TITLE
Cross langage NULL value

### ISSUE THREAD

**Issue by lmichel** [ivoa-std/ModelInstanceInVot#183]
From the Adrian' review

> I think that the serialization should work across languages and should specify the representation of NULL values (vs empty ones).

**Comment by lmichel** [comment]
I propose to add a reserved keyword to indicate the an ATTRIBUTE has a null value.

This value could be applied whatever the datatype is. 
Its meaning in the MIVOT context would be to tell the client: _there in value that can be used to set the value of that attribute_.
The way to process it would remain in charge if the client.

The challenge is to find out a value that does not lead to any confusion:
- classical:  `n.a.`
- specific:  `Mivot-None`
- other: ??

examples

```xml
<ATTRIBUTE dmrole="model.this.role" dmtype="ivoa:bool" value="Mivot-None" \>
<ATTRIBUTE dmrole="model.this.role" dmtype="ivoa:string" value="Mivot-None" \>
```

**Comment by andamian** [comment]
An alternative is to have the attribute optional. If not present, it's considered NULL. Could that work?

**Comment by lmichel** [comment]
This would work, but with a significant impact on the schema since both @value and @ref would be optional.
Let's have a look at this option.

**Comment by andamian** [comment]
Yes, this seems to cover all the cases where the mapping provides the null value and it's very clear what the intention is.

There are also a few ways to represent NULL values in VOTable: 
https://www.ivoa.net/documents/VOTable/20191021/REC-VOTable-1.4-20191021.html#ToC41. 

I don't know if this is mentioned anywhere, but the "XYZ" value should probably follow the VOTable specs 
(in practice, you might want to parse the value with the VOTable parser) which means that there are the VOTable ways to specify a NULL value (NaN for example). Would this be sufficient? Should we have both the VOTable option and the option you described above?

**Comment by lmichel** [comment]
You are right, there are different ways for the VOTable schema to define NULL values.However,  MIVOT does not care about it.
Let's take an example
```xml
<ATTRIBUTE dmrole="model:class.role" dmtype="ivoa:real" ref="_column_3" \>
```
When reading the VOTable through MIVOT, the client will set the attribute `model:class.role` with the value of the column `_column_3`  of the current row. 

 The purpose of MIVOT is to tell the client which value must be taken to populate which attribute.
The way to set the attribute value is totally dependent on the client software environment  (language , inner model..).
If it turns out that the `_column_3`  value is NULL (in the VOTable meaning),  the client has to set the attribute value with NULL, according to its software context.

My opinion  is that MIVOT does not need any specif feature for this, but we may add a little section clarifying this point.

**Comment by lmichel** [comment]
text updated in #185

**Comment by Bonnarel** [comment]
I just read https://github.com/ivoa-std/ModelInstanceInVot/pull/185
I think it's fine. Although not explicit it is the less verbose way to have ATTRIBUTES filled with NULL values.
